test/lint/test_runner/src/main.rs

```diff
@@ -238,6 +245,36 @@ fs:: namespace, which has unsafe filesystem functions marked as deleted.
     }
 }
 
+fn lint_doc_release_note_snippets() -> LintResult {
+    let files =
+        check_output(git().args(["ls-files", "--", "doc/release-notes/release-notes-*.md"]))?
+            .lines()
+            .filter(|file| {
+                let num = file
+                    .trim_start_matches("doc/release-notes/release-notes-")
+                    .trim_end_matches(".md");
+                // If the number is missing a dot, it is a pull request. Thus, the file is a
+                // release note snippet.
+                !num.contains(".")
+            })
+            .map(|file| file.to_string())
+            .collect::<Vec<_>>();
+    if files.is_empty() {
+        Ok(())
+    } else {
+        Err(format!(
+            r#"
+{}
+^^^
+Release note snippets must be put into the doc/ folder directly.
+
+The doc/release-notes/ folder is for archived release notes of previous releases only.
+            "#,
+            files.join("\n")
+        ))
+    }
+}
```

Comment:
I think we can make this more idiomatic Rust by:
* using regex to match the name instead of string manipulations, repeating parts of the path and the file's structure (extension + dot);
* match a slice of the files to return a `Result`;
* extract intermediary results for clarity;
* we could like use [fs::read_dir("doc/release-notes")](https://doc.rust-lang.org/std/fs/fn.read_dir.html) to iterate the folder, but I see that delegating to `git` is common here:
```Rust
fn lint_doc_release_note_snippets() -> LintResult {
    let notes = check_output(git().args(["ls-files", "--", "doc/release-notes"]))?;
    let snippet = Regex::new(r"release-notes-[^.]+\.md$").unwrap();
    let snippet_notes = notes.lines().filter(|f| snippet.is_match(f)).collect::<Vec<_>>();
    match snippet_notes.as_slice() {
        [] => Ok(()),
        _ => Err(format!(r#"
{}
^^^
Release note snippets must be put into the doc/ folder directly.

The doc/release-notes/ folder is for archived release notes of previous releases only.
            "#, snippet_notes.join("\n")))
    }
}
```

Note that this requires adding:
```
[dependencies]
regex = "1.10.6"
```
to the `Cargo.toml` file.

Also note that the `snippet` regex could be pulled out of the method.

Reply:
> Seems overkill to import a dependency to check whether a dot is present in a string view or not

Maybe, in that case we can use glob patterns as well with `ls-files`:

```Rust
fn lint_doc_release_note_snippets() -> LintResult {
    let result = check_output(git().args(["ls-files", "--", "doc/release-notes/release-notes-*.md", ":!doc/release-notes/*.*.md"]))?;
    let snippet_notes = result.lines().collect::<Vec<_>>();
    match snippet_notes.as_slice() {
        [] => Ok(()),
        _ => Err(format!(r#"
{}
^^^
Release note snippets must be put into the doc/ folder directly.

The doc/release-notes/ folder is for archived release notes of previous releases only.
            "#, snippet_notes.join("\n")))
    }
}
```

Reply:
>  think you meant "git pathspec" instead of "shell wildcard patterns"

I did indeed confuse the two in this case, thanks for clarifying.